web/src/settings_components.ts

```diff
@@ -49,7 +75,34 @@ export function get_realm_time_limits_in_minutes(property) {
     return val;
 }
 
-export function get_property_value(property_name, for_realm_default_settings, sub, group) {
+type RealmSetting = Omit<typeof realm, "realm_authentication_methods"> & {
+    realm_authentication_methods: Record<string, boolean>;
```

Comment:
This is somewhat odd. The data received from server in the `realm` object has different value and the one that we used in the app for setting the UI, etc. is different.

Reply:
This is probably related to https://github.com/zulip/zulip/pull/28828.

Reply:
@sahil839 did you figure out a resolution to this detail?

Reply:
Not yet.

I don't think we can change the format returned by the server in `page_params` dict. But we could change `get_property_value` to directly return the `realm.realm_authentication_methods` value and then update the functions calling `get_property_value` to convert the data as needed. Will implement this and see what is the result.